dashboard/config/routes.rb

```diff
@@ -724,8 +724,22 @@
 
     resources :zendesk_session, only: [:index]
 
-    post '/report_abuse', to: 'report_abuse#report_abuse'
-    get '/report_abuse', to: 'report_abuse#report_abuse_form'
+    # Public abuse report form
+    controller :report_abuse do
+      get  '/report_abuse', action: :report_abuse_form
+      post '/report_abuse', action: :report_abuse
+    end
+
+    # partial ports of legacy v3 APIs
+    scope path: '/v3' do
+      controller :report_abuse do
+        get    'channels/:channel_id/abuse', action: :show_abuse
+        delete 'channels/:channel_id/abuse', action: :reset_abuse
+        post   'channels/:channel_id/abuse/delete', action: :reset_abuse
+        patch  '(:endpoint)/:encrypted_channel_id', action: :update_file_abuse,
+               constraints: {endpoint: /(animations|assets|sources|files|libraries)/}
+      end
+    end
```

Comment:
https://github.com/code-dot-org/code-dot-org/blob/bc14a4000a6193fffbcb43c04738a26fa7bc5981/dashboard/config/routes.rb#L623C5-L631C1

Reply:
Similarlly, I followed this example.